privacyidea/lib/auditmodules/base.py

```diff
@@ -164,21 +164,50 @@ def get_total(self, param, AND=True, display_error=True):
         return None
 
     @log_with(log)
-    def log(self, param):  # pragma: no cover
+    def log(self, param):
         """
         This method is used to log the data.
         During a request this method can be called several times to fill the
         internal audit_data dictionary.
+
+        Add new log details in param to the internal log data self.audit_data.
+
+        :param param: Log data that is to be added
+        :type param: dict
+        :return: None
         """
-        pass
+        for k, v in param.items():
+            self.audit_data[k] = v
 
-    def add_to_log(self, param):
+    def add_to_log(self, param, add_with_comma=False):
```

Comment:
Hm, I guess we don't need ``add_with_comma`` anymore now?

Reply:
I thought the other way round! There are a lot of occurrences, where we do ``add_to_log(",somevalue")``  and I thought to use the add_with_comma in these places, since the comma in the paramter does not look that cool.

I have not done it yet, since I thought this would not be part of this PR.

What do you think? I would like to change all these comma-paramters to ``add_with_comma=True``. This would also make the output nicer.
Currently the logic is bad and e.g. info might look like *info1,info2,info3,* with a leading or trailing comma.

Reply:
Oh, I was not aware of the other occurrences where we use ``add_to_log(",somevalue")`` :-) In that case, sure, let's keep it!

Reply:
:+1: